src/crepe/system/CollisionSystem.cpp

```diff
@@ -1,5 +1,381 @@
+			collider1 = std::get<std::reference_wrapper<CircleCollider>>(data1.collider);
+			collider2 = std::get<std::reference_wrapper<CircleCollider>>(data2.collider);
+			break;
+		}
+	}
+	
+	// collision info
+	crepe::CollisionSystem::CollisionInfo collision_info{
+            .first={ collider1, data1.transform, data1.rigidbody },
+            .second={ collider2, data2.transform, data2.rigidbody },
+						.move_back_value = move_back_data.first,
+						.move_back_direction = move_back_data.second,
+        };
+
+	// Determine if static needs to be called
+	determine_collision_handler(collision_info);	
+}
+
+
+std::pair<vec2,CollisionSystem::Direction> CollisionSystem::collision_handler(CollidedInfoStor& data1,CollidedInfoStor& data2,ColliderStorType type) {
+	vec2 move_back;
+	switch (type) {
+		case ColliderStorType::BOX_BOX:	{
+			const BoxCollider & collider1 = std::get<std::reference_wrapper<BoxCollider>>(data1.collider);
+			const BoxCollider & collider2 = std::get<std::reference_wrapper<BoxCollider>>(data2.collider);
+			vec2 collider_pos1 = current_position(collider1.offset, data1.transform, data1.rigidbody);
+			vec2 collider_pos2 = current_position(collider2.offset, data2.transform, data2.rigidbody);
+			move_back = box_box_move_back(collider1,collider2,collider_pos1,collider_pos2);
+		}
+		case ColliderStorType::BOX_CIRCLE: {
+		
+		}
+		case ColliderStorType::CIRCLE_CIRCLE:	{
+		
+		}
+		case ColliderStorType::CIRCLE_BOX:	{
+		
+		}
+	}
+
+	Direction move_back_direction = Direction::NONE;
+	if(move_back.x != 0 && move_back.y > 0) {
+		move_back_direction = Direction::BOTH;
+	} else if (move_back.x != 0) {
+		move_back_direction = Direction::X_DIRECTION;
+		if(data1.rigidbody.data.linear_velocity.y != 0)	
+		move_back.y = data1.rigidbody.data.linear_velocity.y * (move_back.x/data1.rigidbody.data.linear_velocity.x);
+	} else if (move_back.y != 0) {
+		move_back_direction = Direction::Y_DIRECTION;
+		if(data1.rigidbody.data.linear_velocity.x != 0) 
+		move_back.x = data1.rigidbody.data.linear_velocity.x * (move_back.y/data1.rigidbody.data.linear_velocity.y);
+	}
+
+	return {move_back,move_back_direction};
+}
+
+vec2 CollisionSystem::box_box_move_back(const BoxCollider& box_collider1,const BoxCollider& box_collider2,vec2 final_position1,vec2 final_position2)
+{
+	vec2 resolution; // Default resolution vector
+	vec2 delta = final_position2 - final_position1;
+
+	// Compute half-dimensions of the boxes
+	float half_width1 = box_collider1.width / 2.0;
+	float half_height1 = box_collider1.height / 2.0;
+	float half_width2 = box_collider2.width / 2.0;
+	float half_height2 = box_collider2.height / 2.0;
+
+	// Calculate overlaps along X and Y axes
+	float overlap_x = (half_width1 + half_width2) - std::abs(delta.x);
+	float overlap_y = (half_height1 + half_height2) - std::abs(delta.y);
+
+	// Check if there is a collision
+	if (overlap_x > 0 && overlap_y > 0) {//should always be true check if this can be removed
```

Comment:
moet deze weggehaald worden er is als het goed is toch altijd een collision als je hier bent gekomen?

Reply:
Dat staat er ook bij. Dit moet ik nog testen.